sdc11073/pmtypes.py

```diff
@@ -60,11 +60,15 @@ def _sortedContainerProperties(self):
 
 
     def __eq__(self, other):
-        ''' compares all properties'''
+        """ compares all properties"""
         try:
             for name, dummy in self._sortedContainerProperties():
-                if getattr(self, name) == getattr(other, name):
-                    # only __eq__ methods implemented, therefore compare for equality
+                my_value = getattr(self, name)
+                other_value = getattr(other, name)
+                if my_value == other_value:
+                    continue
+                elif (isinstance(my_value, Decimal) or isinstance(other_value, Decimal)) and isclose(my_value, other_value):
```

Comment:
Because e.g. Decimal('1.1') is not equal to 1.1! float(1.1) has no exact representation, it is something like 
1.100000000000000088817841970012523233890533447265625.

Reply:
True, but it also allows two Decimals to pass if they're close, not equal, and it seems odd to use the precise datatype and then allow an imprecise comparison.